ARCs/arc-0014.md

```diff
@@ -0,0 +1,67 @@
+---
+ams: 14
+title: Algorand standard for Authentication
+status: Draft
+---
+
+# Algorand Standard for authentication
+
+## Summary
+
+This document introduces the standard for authentication using the algorand accounts.
+
+## Abstract
+
+The goal of this standard is to define authorization process and authentication procedures for communication between web application and backend services or between two backend services. 
+
+## Specification
+
+The key words "**MUST**", "**MUST NOT**", "**REQUIRED**", "**SHALL**", "**SHALL NOT**", "**SHOULD**", "**SHOULD NOT**", "**RECOMMENDED**", "**MAY**", and "**OPTIONAL**" in this document are to be interpreted as described in [RFC-2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+Base64 is the standard described in [RFC-4648](https://www.ietf.org/rfc/rfc4648.txt).
+
+## Request for comments
+
+Please comment or suggest pull request for this document here: [https://github.com/scholtz/AMS/issues](https://github.com/scholtz/AMS/issues)
+
+### Authentication
+
+Authentication is process in which person generates the authorization token for communication with backend service.
+
+Token MUST be generated from Algorand public and private key - from algorand address.
+
+In the first step of the authentication, the message to be signed is generated. The message MUST contain note field with configurable Realm. Realm is identifier of the service. It MAY be the web address of the service for intended purpose.
+
+The message MUST be self signed - the receiver is the same as the sender.
```

Comment:
lets define this as a pay transaction, of 0 amount, from the authenticator to the authenticator (or a better word?) with the note field populated with the above and the rest of the parameters below. like a template transaction 
```
txn {
...
}
```

Reply:
i would not do it as a must.. the authenticator does not have to have wish to reveal his address. also he might want to use it in different domains.. eg google authentication service is on one domain, and it serves multiple other hosts backends.. the only thing that the backend has to verify when writing some user data that the realm is within its scope.. if the realm will be in the note or as a receiver, it is up to the user. if you guys believe we should change MUST to SHOULD go on..